### PR TITLE
CB-20093 Add argument runAutoConfig for applyHostTemplate call to CM

### PR DESCRIPTION
**Pull request by doktoric**
See detailed description in the commit message.

### CODE REVIEW

**Comment by doktoric** [comment]
resetest this please

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java [hunk 1]
```diff
@@ -177,15 +179,20 @@ public void initApiClient() throws ClusterClientInitException {
...
         try {
-            apiClient = clouderaManagerApiClientProvider.getV31Client(stack.getGatewayPort(), user, password, clientConfig);
+            v31Client = clouderaManagerApiClientProvider.getV31Client(stack.getGatewayPort(), user, password, clientConfig);
         } catch (ClouderaManagerClientInitException e) {
             throw new ClusterClientInitException(e);
         }
+        try {
+            v52Client = clouderaManagerApiClientProvider.getV52Client(stack.getGatewayPort(), user, password, clientConfig);
+        } catch (ClouderaManagerClientInitException e) {
+            LOGGER.warn("Client init failed for V45 client!");
```
Comment (lajosrodek) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java:189]: Nit: `V45` -> `V52`

File: cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationServiceTest.java
Comment (lajosrodek) [inline cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationServiceTest.java:1]: Could you please add a test case for the new scenario?

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationServiceTest.java
Comment (lajosrodek) [reply cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationServiceTest.java:1]: Still pending. :smile:

**Comment by doktoric** [comment]
retest this please

**Pull request merged**